aip/general/0236.md

```diff
@@ -0,0 +1,355 @@
+- Previewing a change to a current policy (update or delete use-cases) **should** require copying the full payload of the live resource into the inlined policy field and making changes to fields-under-test as needed for preview. Delete requires setting rules collection to null.
+  - The name inside “policy” **must** be the name of the parent.
+- Previewing the creation of a brand new policy **should** be supported. 
+  - If the Control Plane does not support a nested collection without a parent, the user **should** create a top-level resource which has the correct rules to represent a No-Op. Then, experiments can be added underneath that resource.
+- If multiple previews are supported, calling the create more than once on a resource **should** lead to the creation of multiple experiments represented as new subresources until the quota is reached.
+- If the operation failed, return the appropriate error code ([AIP-193][aip-193]).
+
+[aip-133]: https://aip.dev/133
+[aip-193]: https://aip.dev/193
+
+#### update
+
+```proto
+rpc UpdatePolicyExperiment(UpdatePolicyExperimentRequest) returns (PolicyExperiment);
+```
+- Standard methods: Update ([AIP-134]) **must** be used to change the fields-under-test in an experiment used for preview mode.
+  - The name inside “policy” **must not** be changed.
+- The Control Plane **should** set the PreviewMetadata.state to INACTIVE if the state was ACTIVE at the time of an update. This is so the user can easily distinguish between previews of different fields-under-test.
+- If the operation failed, the old experiment **must** be returned alongside the appropriate error code ([AIP-193][aip-193]).
+
+[aip-134]: https://aip.dev/134
+
+#### get
+
+```proto
+rpc GetPolicyExperiment(GetPolicyExperimentRequest) returns (PolicyExperiment);
+```
+- Standard methods: Get ([AIP-131][aip-131]) **must** be used to get the experiment used for preview mode.
+
+[aip-131]: https://aip.dev/131
+
+#### list
+
+```proto
+rpc ListPolicyExperiments(ListPolicyExperimentsRequest) returns (ListPolicyExperimentsResponse);
+```
+- Standard methods: List ([AIP-132][aip-132]) **should** be used to list experiments for preview mode.
+- List **should** support Filtering ([AIP-160][aip-160]) semantics.
+- Filtering on PreviewMetadata.state **should** indicate which experiments are actively being previewed or not.
+
+[aip-132]: https://aip.dev/132
+[aip-160]: https://aip.dev/160
+
+#### delete
+
+```proto
+rpc DeletePolicyExperiment(DeletePolicyExperimentRequest) returns (google.protobuf.Empty);
+```
+- Standard methods: Delete ([AIP-135][aip-135]) **must** be used to delete experiments.
+
+[aip-135]: https://aip.dev/135
+
+### Parent Resource API Changes
+
+#### delete
+
+- A delete of the top-level resource **must** delete all experiments.
+- To maintain the experiments, the top-level Policy **must** be changed to a no-op Policy instead of using delete.
+
+### Custom Method APIs
+
+#### startPreview
```

Comment:
If there is one meaningful way to start an experiment, consider the shorter custom method name "start"

That also aligns the naming convention between the "start" custom method with the "start_time" output only field.

If additional ways to start an experiment come later, they may be added as an argument enum on the StartPolicyExperimentRequest message, or field on the experiment.

Reply:
I don't think this is the right idea as experiment can be used for other use-cases like progressive rollout. At least for our internal use-cases.